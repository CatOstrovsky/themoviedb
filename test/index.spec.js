import dotenv from 'dotenv';
import OpenVpn from '../src';

dotenv.config();

describe('basic', () => {
  it('should setup()', async () => {
    const vpn = new OpenVpn();
    // console.log(vpn);
    const popular = await vpn.miscPopularMovies();
    expect(popular).toMatchSnapshot();
  });
});
