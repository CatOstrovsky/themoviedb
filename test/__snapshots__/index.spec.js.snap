// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should setup() 1`] = `
Object {
  "page": 1,
  "results": Array [
    Object {
      "adult": false,
      "backdrop_path": "/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg",
      "genre_ids": Array [
        878,
      ],
      "id": 335983,
      "original_language": "en",
      "original_title": "Venom",
      "overview": "Eddie Brock is a reporter—investigating people who want to go unnoticed. But after he makes a terrible discovery at the Life Foundation, he begins to transform into ‘Venom’.  The Foundation has discovered creatures called symbiotes, and believes they’re the key to the next step in human evolution. Unwittingly bonded with one, Eddie discovers he has incredible new abilities—and a voice in his head that’s telling him to embrace the darkness.",
      "popularity": 434.485,
      "poster_path": "/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg",
      "release_date": "2018-10-03",
      "title": "Venom",
      "video": false,
      "vote_average": 6.5,
      "vote_count": 2868,
    },
    Object {
      "adult": false,
      "backdrop_path": "/blv65adWsJQZ5Iog0OU5opVf6Oa.jpg",
      "genre_ids": Array [
        28,
        35,
      ],
      "id": 523873,
      "original_language": "zh",
      "original_title": "功夫联盟",
      "overview": "Martial arts comedy following a group of kung fu legends banding together to take on the bad guys. The legends includes VINCENT ZHAO reprising his role as ‘Wong Fei Hung’ with DENNIS TO once again portraying ‘Wing Chun’ master ‘Ip Man’, DANNY CHAN KWOK KWAN as ‘Chen Zhen’ and ANDY ON as master ‘Huo Yuan Jia’.",
      "popularity": 201.487,
      "poster_path": "/rW0A73hjzPWVwADlCTLnjLhAFLX.jpg",
      "release_date": "2018-10-19",
      "title": "Kung Fu League",
      "video": false,
      "vote_average": 4.3,
      "vote_count": 47,
    },
    Object {
      "adult": false,
      "backdrop_path": "/xgbeBCjmFpRYHDF7tQ7U98EREWp.jpg",
      "genre_ids": Array [
        10751,
        14,
        12,
      ],
      "id": 338952,
      "original_language": "en",
      "original_title": "Fantastic Beasts: The Crimes of Grindelwald",
      "overview": "Gellert Grindelwald has escaped imprisonment and has begun gathering followers to his cause—elevating wizards above all non-magical beings. The only one capable of putting a stop to him is the wizard he once called his closest friend, Albus Dumbledore. However, Dumbledore will need to seek help from the wizard who had thwarted Grindelwald once before, his former student Newt Scamander, who agrees to help, unaware of the dangers that lie ahead. Lines are drawn as love and loyalty are tested, even among the truest friends and family, in an increasingly divided wizarding world.",
      "popularity": 193.051,
      "poster_path": "/uyJgTzAsp3Za2TaPiZt2yaKYRIR.jpg",
      "release_date": "2018-11-14",
      "title": "Fantastic Beasts: The Crimes of Grindelwald",
      "video": false,
      "vote_average": 7,
      "vote_count": 1900,
    },
    Object {
      "adult": false,
      "backdrop_path": "/AuA50D7T7S7OEVcGo0ZKaMhERn0.jpg",
      "genre_ids": Array [
        12,
        28,
      ],
      "id": 375588,
      "original_language": "en",
      "original_title": "Robin Hood",
      "overview": "A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.",
      "popularity": 177.875,
      "poster_path": "/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg",
      "release_date": "2018-11-20",
      "title": "Robin Hood",
      "video": false,
      "vote_average": 6,
      "vote_count": 194,
    },
    Object {
      "adult": false,
      "backdrop_path": "/uKwOX7MtKlAaGeCQe6c4jc1vZpj.jpg",
      "genre_ids": Array [
        28,
        12,
        14,
        16,
      ],
      "id": 507569,
      "original_language": "ja",
      "original_title": "劇場版 七つの大罪 天空の囚われ人",
      "overview": "Traveling in search of the rare ingredient, “sky fish”  Meliodas and Hawk arrive at a palace that floats above the clouds. The people there are busy preparing a ceremony, meant to protect their home from a ferocious beast that awakens once every 3,000 years. But before the ritual is complete, the Six Knights of Black—a Demon Clan army—removes the seal on the beast, threatening the lives of everyone in the Sky Palace.",
      "popularity": 165.911,
      "poster_path": "/r6pPUVUKU5eIpYj4oEzidk5ZibB.jpg",
      "release_date": "2018-08-18",
      "title": "The Seven Deadly Sins: Prisoners of the Sky",
      "video": false,
      "vote_average": 5.9,
      "vote_count": 97,
    },
    Object {
      "adult": false,
      "backdrop_path": "/pbXgLEYh8rlG2Km5IGZPnhcnuSz.jpg",
      "genre_ids": Array [
        18,
        10402,
      ],
      "id": 424694,
      "original_language": "en",
      "original_title": "Bohemian Rhapsody",
      "overview": "Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet – finding a way to keep the band together amid the success and excess.",
      "popularity": 153.252,
      "poster_path": "/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg",
      "release_date": "2018-10-24",
      "title": "Bohemian Rhapsody",
      "video": false,
      "vote_average": 8.2,
      "vote_count": 2061,
    },
    Object {
      "adult": false,
      "backdrop_path": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
      "genre_ids": Array [
        12,
        28,
        14,
      ],
      "id": 299536,
      "original_language": "en",
      "original_title": "Avengers: Infinity War",
      "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
      "popularity": 135.682,
      "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
      "release_date": "2018-04-25",
      "title": "Avengers: Infinity War",
      "video": false,
      "vote_average": 8.3,
      "vote_count": 9751,
    },
    Object {
      "adult": false,
      "backdrop_path": "/ivhK1NjnFji6lD5dkAThSzd3Mgq.jpg",
      "genre_ids": Array [
        10751,
        16,
        35,
      ],
      "id": 404368,
      "original_language": "en",
      "original_title": "Ralph Breaks the Internet",
      "overview": "Taking place six years following the events of the first film, the story will center on Ralph's adventures in the Internet data space when a Wi-Fi router gets plugged into the arcade as he must find a replacement part to fix Sugar Rush.",
      "popularity": 128.706,
      "poster_path": "/m110vLaDDOCca4hfOcS5mK5cDke.jpg",
      "release_date": "2018-11-20",
      "title": "Ralph Breaks the Internet",
      "video": false,
      "vote_average": 7.5,
      "vote_count": 162,
    },
    Object {
      "adult": false,
      "backdrop_path": "/aw4FOsWr2FY373nKSxbpNi3fz4F.jpg",
      "genre_ids": Array [
        28,
        53,
        12,
      ],
      "id": 353081,
      "original_language": "en",
      "original_title": "Mission: Impossible - Fallout",
      "overview": "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfill his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
      "popularity": 123.467,
      "poster_path": "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
      "release_date": "2018-07-13",
      "title": "Mission: Impossible - Fallout",
      "video": false,
      "vote_average": 7.2,
      "vote_count": 2502,
    },
    Object {
      "adult": false,
      "backdrop_path": "/eW2PIUIyagoYgL1Q7nu0E6EMnB3.jpg",
      "genre_ids": Array [
        35,
        16,
        10751,
        12,
        14,
      ],
      "id": 446894,
      "original_language": "en",
      "original_title": "Smallfoot",
      "overview": "A bright young yeti finds something he thought didn't exist—a human. News of this “smallfoot” throws the simple yeti community into an uproar over what else might be out there in the big world beyond their snowy village.",
      "popularity": 123.051,
      "poster_path": "/zfaiO7QgpcvR8XDOMokWLRfKeTE.jpg",
      "release_date": "2018-09-20",
      "title": "Smallfoot",
      "video": false,
      "vote_average": 6.6,
      "vote_count": 235,
    },
    Object {
      "adult": false,
      "backdrop_path": "/f4E0ocYeToEuXvczZv6QArrMDJ.jpg",
      "genre_ids": Array [
        878,
        28,
        53,
        12,
      ],
      "id": 346910,
      "original_language": "en",
      "original_title": "The Predator",
      "overview": "When a kid accidentally triggers the universe's most lethal hunters' return to Earth, only a ragtag crew of ex-soldiers and a disgruntled female scientist can prevent the end of the human race.",
      "popularity": 121.476,
      "poster_path": "/wMq9kQXTeQCHUZOG4fAe5cAxyUA.jpg",
      "release_date": "2018-09-13",
      "title": "The Predator",
      "video": false,
      "vote_average": 5.2,
      "vote_count": 1072,
    },
    Object {
      "adult": false,
      "backdrop_path": "/juc9wt7Eh2IarLL5S1yQ1a21O1A.jpg",
      "genre_ids": Array [
        16,
        10751,
        35,
        14,
      ],
      "id": 360920,
      "original_language": "en",
      "original_title": "The Grinch",
      "overview": "The Grinch hatches a scheme to ruin Christmas when the residents of Whoville plan their annual holiday celebration.",
      "popularity": 112.831,
      "poster_path": "/lCfKAAFyANPFn04xOPqazd8jwnc.jpg",
      "release_date": "2018-11-08",
      "title": "The Grinch",
      "video": false,
      "vote_average": 6.3,
      "vote_count": 299,
    },
    Object {
      "adult": false,
      "backdrop_path": "/rxYG6Sj95as9rv9wKIHUx6ATWd3.jpg",
      "genre_ids": Array [
        878,
      ],
      "id": 428078,
      "original_language": "en",
      "original_title": "Mortal Engines",
      "overview": "Set in a world many thousands of years in the future. Earth’s cities now roam the globe on huge wheels, devouring each other in a struggle for ever diminishing resources. On one of these massive Traction Cities, Tom Natsworthy has an unexpected encounter with a mysterious young woman from the Outlands who will change the course of his life forever.",
      "popularity": 106.702,
      "poster_path": "/uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg",
      "release_date": "2018-12-06",
      "title": "Mortal Engines",
      "video": false,
      "vote_average": 5.8,
      "vote_count": 22,
    },
    Object {
      "adult": false,
      "backdrop_path": "/6P3c80EOm7BodndGBUAJHHsHKrp.jpg",
      "genre_ids": Array [
        28,
        12,
        878,
        35,
        10751,
      ],
      "id": 363088,
      "original_language": "en",
      "original_title": "Ant-Man and the Wasp",
      "overview": "Just when his time under house arrest is about to end, Scott Lang puts again his freedom at risk to help Hope van Dyne and Dr. Hank Pym dive into the quantum realm and try to accomplish, against time and any chance of success, a very dangerous rescue mission.",
      "popularity": 93.148,
      "poster_path": "/rv1AWImgx386ULjcf62VYaW8zSt.jpg",
      "release_date": "2018-07-04",
      "title": "Ant-Man and the Wasp",
      "video": false,
      "vote_average": 6.9,
      "vote_count": 3673,
    },
    Object {
      "adult": false,
      "backdrop_path": "/fgsHxz21B27hOOqQBiw9L6yWcM7.jpg",
      "genre_ids": Array [
        27,
        9648,
        53,
      ],
      "id": 439079,
      "original_language": "en",
      "original_title": "The Nun",
      "overview": "When a young nun at a cloistered abbey in Romania takes her own life, a priest with a haunted past and a novitiate on the threshold of her final vows are sent by the Vatican to investigate. Together they uncover the order’s unholy secret. Risking not only their lives but their faith and their very souls, they confront a malevolent force in the form of the same demonic nun that first terrorized audiences in “The Conjuring 2” as the abbey becomes a horrific battleground between the living and the damned.",
      "popularity": 91.545,
      "poster_path": "/sFC1ElvoKGdHJIWRpNB3xWJ9lJA.jpg",
      "release_date": "2018-09-05",
      "title": "The Nun",
      "video": false,
      "vote_average": 5.7,
      "vote_count": 1697,
    },
    Object {
      "adult": false,
      "backdrop_path": "/2uSCHUsmzb6KkQPFSxBQ7bgfJLE.jpg",
      "genre_ids": Array [
        28,
        878,
        53,
      ],
      "id": 345940,
      "original_language": "en",
      "original_title": "The Meg",
      "overview": "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
      "popularity": 80.922,
      "poster_path": "/eyWICPcxOuTcDDDbTMOZawoOn8d.jpg",
      "release_date": "2018-08-09",
      "title": "The Meg",
      "video": false,
      "vote_average": 5.9,
      "vote_count": 1837,
    },
    Object {
      "adult": false,
      "backdrop_path": "/mabuNsGJgRuCTuGqjFkWe1xdu19.jpg",
      "genre_ids": Array [
        28,
        12,
        16,
        10751,
      ],
      "id": 260513,
      "original_language": "en",
      "original_title": "Incredibles 2",
      "overview": "Elastigirl springs into action to save the day, while Mr. Incredible faces his greatest challenge yet – taking care of the problems of his three children.",
      "popularity": 79.043,
      "poster_path": "/x1txcDXkcM65gl7w20PwYSxAYah.jpg",
      "release_date": "2018-06-14",
      "title": "Incredibles 2",
      "video": false,
      "vote_average": 7.6,
      "vote_count": 4010,
    },
    Object {
      "adult": false,
      "backdrop_path": "/crn09DUYgmFUPbOroxt1tQSoi0r.jpg",
      "genre_ids": Array [
        53,
        27,
      ],
      "id": 424139,
      "original_language": "en",
      "original_title": "Halloween",
      "overview": "Laurie Strode comes to her final confrontation with Michael Myers, the masked figure who has haunted her since she narrowly escaped his killing spree on Halloween night four decades ago.",
      "popularity": 78.801,
      "poster_path": "/bXs0zkv2iGVViZEy78teg2ycDBm.jpg",
      "release_date": "2018-10-18",
      "title": "Halloween",
      "video": false,
      "vote_average": 6.4,
      "vote_count": 1033,
    },
    Object {
      "adult": false,
      "backdrop_path": "/meyZ0ZZLLNaKIrvbmVpuXzZ11sD.jpg",
      "genre_ids": Array [
        27,
        14,
        53,
        10751,
        9648,
      ],
      "id": 463821,
      "original_language": "en",
      "original_title": "The House with a Clock in Its Walls",
      "overview": "Lewis Barnavelt, a shy but indomitable ten-year-old orphan, goes to live with his uncle Jonathan, a former entertaining magician, who hides too many creepy secrets, in a creaky old house with a mysterious tick-tocking heart which Jonathan is obsessed to find with the assistance of his enigmatic next-door neighbor Mrs. Zimmerman.",
      "popularity": 75.424,
      "poster_path": "/qM66Hv4ByAxnilr0jaqCA9uOD4Y.jpg",
      "release_date": "2018-09-15",
      "title": "The House with a Clock in Its Walls",
      "video": false,
      "vote_average": 6.2,
      "vote_count": 366,
    },
    Object {
      "adult": false,
      "backdrop_path": "/a24i4Qw6qFTtkFLQsncv8WPwevj.jpg",
      "genre_ids": Array [
        53,
        28,
        80,
      ],
      "id": 345887,
      "original_language": "en",
      "original_title": "The Equalizer 2",
      "overview": "Robert McCall, who serves an unflinching justice for the exploited and oppressed, embarks on a relentless, globe-trotting quest for vengeance when a long-time girl friend is murdered.",
      "popularity": 74.982,
      "poster_path": "/cQvc9N6JiMVKqol3wcYrGshsIdZ.jpg",
      "release_date": "2018-07-19",
      "title": "The Equalizer 2",
      "video": false,
      "vote_average": 6.3,
      "vote_count": 1204,
    },
  ],
  "total_pages": 990,
  "total_results": 19790,
}
`;
